Gems/ROS2/Code/Include/ROS2/RobotControl/ControlConfiguration.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Component/Entity.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ROS2
+{
+    //! Configuration for handling of robot control buses.
+    //! Used through ROS2RobotControlComponent.
+    struct ControlConfiguration
+    {
+    public:
+        AZ_TYPE_INFO(ControlConfiguration, "{3D3E69EE-0F28-46D5-95F1-956550BA97B9}");
+
+        //! Type of control for the robot.
+        //! Different types of steering can fit different platforms,
+        //! depending on the type their mobile base (four wheels, omniwheels, ..).
+        enum Steering
+        {
+            Twist, //!< @see <a href="https://docs.ros2.org/latest/api/geometry_msgs/msg/Twist.html">Twist</a>.
+            Ackermann //!< @see <a href="https://github.com/ros-drivers/ackermann_msgs/blob/ros2/msg/AckermannDrive.msg">AckermannDrive</a>.
+        };
```

Comment:
Scoped enums are prefered to unscoped enums per the coding standard: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#enums

This is also important because the underlying type of C style `enum Steering` is implementation defined because it is not specified
This means that because m_steering variable being serialized on Linux might be an `unsigned int`, but if it was loaded on a Windows it could be interpreted as a `signed int`.

Therefore it is always important to at least specify an underlying type when using an unscoped enum that is being serialized.

For scoped enums using `enum class`, there underlying type is by default `int`, so the serialization issue doesn't present itself across platforms in that case.